
# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1652358904 14400
# Node ID 6e91c61f626f3067640e6f55fcf2f646031313db
# Parent  4f2f81989d50b760046364037b816bd794f8f3e2
pkg: Avoid error when unlinking non-existent files (bug #62436).

* scripts/pkg/private/build.m: Check configure and Makefile exist before trying
to unlink them.

diff --git a/scripts/pkg/private/build.m b/scripts/pkg/private/build.m
--- a/scripts/pkg/private/build.m
+++ b/scripts/pkg/private/build.m
@@ -77,8 +77,12 @@
     else
       arch_abi = getarch ();
       configure_make (desc, build_root, verbose);
-      unlink (fullfile (build_root, "src", "configure"));
-      unlink (fullfile (build_root, "src", "Makefile"));
+      if exist (fullfile (build_root, "src", "configure"), "file")
+        unlink (fullfile (build_root, "src", "configure"));
+      endif
+      if exist (fullfile (build_root, "src", "Makefile"), "file")
+        unlink (fullfile (build_root, "src", "Makefile"));
+      endif
     endif
     tar_name = [desc.name "-" desc.version "-" arch_abi ".tar"];
     tar_path = fullfile (builddir, tar_name);

